Orange/widgets/utils/tests/test_test_run.py

```diff
@@ -0,0 +1,90 @@
+import sys
+import unittest
+from unittest.mock import MagicMock
+
+from AnyQt.QtWidgets import QApplication
+
+from Orange.widgets.widget import OWWidget, Input
+from Orange.widgets.utils import test_run
+
+# prevent test_run from calling sys.exit
+sys.exit = MagicMock()
+
+# prevent multiple initializations of QApplication
+app = QApplication([])
+test_run.QApplication = lambda *_: app
+
+# prevent entering the application's main loop
+app.exec_ = MagicMock()
```

Comment:
Move all of the above to `setUpClass `and then undo the modification s in `tearDownClass`.

Reply:
Moved. `app = QApplication([])` remains outside, I suppose?